server/hedley/modules/custom/hedley_reports/hedley_reports.module

```diff
@@ -3416,6 +3407,969 @@ function hedley_reports_count_acute_illness_signs(array $values) {
+    'well_child_opv_immunisation' => 'u',
+    'well_child_pcv13_immunisation' => 'v',
+    'well_child_photo' => 'w',
+    'well_child_pregnancy_summary' => 'x',
+    'well_child_rotarix_immunisation' => 'y',
+    'well_child_send_to_hc' => 'z',
+    'well_child_symptoms_review' => '1',
+    'well_child_vitals' => '2',
+    'well_child_vitamin_a' => '3',
+    'well_child_weight' => '4',
+    'well_child_hpv_immunisation' => '5',
+    'well_child_dtp_sa_immunisation' => '6',
+  ];
+
+  // Load all encounters of current participant.
+  $query = new EntityFieldQuery();
+  $result = $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'well_child_encounter')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->fieldCondition('field_individual_participant', 'target_id', $participant->nid)
+    ->propertyOrderBy('nid')
+    ->execute();
+
+  if (empty($result['node'])) {
+    return;
+  }
+
+  // Get birthdate.
+  $person_id = $participant->field_person[LANGUAGE_NONE][0]['target_id'];
+  try {
+    $person = node_load($person_id);
+    $birth_date = explode(' ', $person->field_birth_date[LANGUAGE_NONE][0]['value'])[0];
+    $birth_date_obj = new DateTime($birth_date);
+    $gender = $person->field_gender[LANGUAGE_NONE][0]['value'];
+  }
+  catch (Exception $e) {
+    $birth_date_obj = NULL;
+    $gender = 'male';
+  }
+
+  // Encounters are sorted ASC.
+  $encounters = node_load_multiple(array_keys($result['node']));
+  $encounters_data = [];
+  foreach ($encounters as $encounter) {
+    // Skip encounter if exclusion flag is raised and it's report data is set.
+    if ($exclude_set && !empty($encounter->field_reports_data[LANGUAGE_NONE][0]['value'])) {
+      continue;
+    }
+
+    // Loading all measurements that belong to encounter.
+    $measurements_ids = hedley_reports_load_individual_encounter_measurements_ids($encounter);
+    // If encounter type field is empty, we default to pediatric care.
+    if (empty($encounter->field_well_child_encounter_type)) {
+      $encounter_type = 'pediatric-care';
+    }
+    else {
+      $encounter_type = $encounter->field_well_child_encounter_type[LANGUAGE_NONE][0]['value'];
+    }
+
+    $measurements = !empty($measurements_ids) ? node_load_multiple($measurements_ids) : [];
+    // Ordering measurements by type.
+    $measurements_by_type = [];
+    foreach ($measurements as $measurement) {
+      $measurements_by_type[$measurement->type] = $measurement;
+    }
+
+    $encounters_data[] = [
+      'encounter' => $encounter,
+      'encounter_type' => $encounter_type,
+      'measurements_by_type' => $measurements_by_type,
+    ];
+  }
+
+  foreach ($encounters_data as $index => $encounter_data) {
+    $encounter = $encounter_data['encounter'];
+    $measurements_by_type = $encounter_data['measurements_by_type'];
+
+    // Resolve encounter start date.
+    $start_date = explode(' ', $encounter->field_scheduled_date[LANGUAGE_NONE][0]['value'])[0];
```

Comment:
what do we do here? Extracting year or dropping time component? `DateTime` would likely do a same. Just thinking.

Reply:
Extracting YYYY-MM-DD and dropping HH-MM.